app/adapters/alma_adapter.rb

```diff
@@ -103,4 +107,59 @@ def perm_location(bib_item)
     def apikey
       Rails.configuration.alma[:read_only_apikey]
     end
+
+    # @param mms_id [String]. e.g id = "991227850000541"
+    # @param all_holding_ids [Array<String>]
+    # @return [Hash<String, Array>] hash with holding id as key and notes as values
+    def bib_item_holding_notes(mms_id, all_holding_ids)
+      notes_from_bib_record = holding_notes_from_bib_record(mms_id)
+
+      # Get list of holding ids not included in bib record AVA fields
+      missing_holding_ids = all_holding_ids - notes_from_bib_record.keys
+
+      # Bib record AVA fields do not include the holding ID for temporary holding locations.
+      # We have to fetch notes for these holdings individually using their 866
+      # value.
+      notes_from_holdings = holding_notes_from_holding_records(mms_id, missing_holding_ids)
+
+      # Return all notes
+      notes_from_bib_record.merge(notes_from_holdings)
+    end
+
+    # Get notes from AVA subfields on bib record
+    # @param mms_id [String]. e.g id = "991227850000541"
+    # @param [Hash<String, Array>] hash with holding id as key and notes as values
+    def holding_notes_from_bib_record(mms_id)
+      record = get_bib_record(mms_id)
+      return {} unless record&.try(:fields)
+      notes_by_holding = {}
+      ava_subfields = record.fields.select { |f| f.tag == "AVA" }.select { |s| s.codes.include? "t" }
+      ava_subfields.each do |subfield|
+        holding_id = subfield["8"]
+        note = subfield["t"]
+        if notes_by_holding.key?(holding_id)
```

Comment:
Could this conditional be collapsed by using fetch? `notes_by_holding[holding_id] = notes_by_holding.fetch(holding_id, []) << note`

Reply:
That's a nice refactor. I'll give it a try.